### ISSUE TITLE
Add missing ARIA attr/prop reflections

### ISSUE THREAD

**Issue by nolanlawson** [salesforce/lwc#3693]
Looking at [the ARIAMixin spec](https://w3c.github.io/aria/#ARIAMixin) and the `aria*` props available on `Element.prototype` in Chrome, there are a few we're missing.

Missing per Chrome:

- ariaBrailleLabel
- ariaBrailleRoleDescription
- ariaDescription

Missing per the latest spec:

- ariaColIndexText
- ariaDescription
- ariaRowIndexText

Once https://github.com/salesforce/lwc/pull/3666 is released and we are out of the business of applying global polyfills, it should be safe to add these to the `LightningElement`/`BaseBridgeElement` prototypes. This will make it so that a template like:

```html
<x-cmp aria-description="foo"></x-cmp>
```

...results in the `ariaDescription` prop (not attr) being set on `x-cmp`. This is consistent with what we currently do for most ARIA props, which allows for things like overriding the `aria*` getter/setter, plus convenience accessors like `this.aria*`.

Note: per Safari we are also missing the `*Element`/`*Elements` APIs, but since these are element reflection and not string reflection, we probably shouldn't do anything with these. Also we already have our non-standard string reflection for these:

- ariaActiveDescendantElement
- ariaControlsElements
- ariaDescribedByElements
- ariaDetailsElements
- ariaErrorMessageElements
- ariaFlowToElements
- ariaLabelledByElements
- ariaOwnsElements

**Comment by MamataLV** [comment]
Hi @nolanlawson,
Is this issue open to fix? And can I work on this?

**Comment by nolanlawson** [comment]
@MamataLV Yes, but note some uncertainties: https://github.com/salesforce/lwc/pull/3702#issuecomment-1720253273

**Comment by nolanlawson** [comment]
[Firefox is shipping ARIA string reflection in Nightly](https://bugzilla.mozilla.org/show_bug.cgi?id=1785412). I just checked, and here are the props they support that we don't:

- ariaColIndexText
- ariaDescription
- ariaRowIndexText

This matches the missing props from the spec.

**Comment by nolanlawson** [comment]
Fixed by #3702
